test/models/repository.test.js

```diff
@@ -654,6 +766,40 @@ describe('Repository', function() {
         `,
       );
     });
+
+    it('records an event', async function() {
+      const workingDirPath = await cloneRepository('multiple-commits');
+      const repo = new Repository(workingDirPath);
+      await repo.getLoadPromise();
+
+      sinon.stub(reporterProxy, 'addEvent');
+
+      assert.deepEqual(reporterProxy.addEvent.callCount, 0);
+      await repo.undoLastCommit();
+      assert.deepEqual(reporterProxy.addEvent.callCount, 1);
+
+      const args = reporterProxy.addEvent.lastCall.args;
+      assert.deepEqual(args[0], 'undo-last-commit');
+      assert.deepEqual(args[1], {package: 'github'});
+    });
+
+    it('does not record an event if operation fails', async function() {
+      const workingDirPath = await cloneRepository('multiple-commits');
+      const repo = new Repository(workingDirPath);
+      await repo.getLoadPromise();
+
+      sinon.stub(reporterProxy, 'addEvent');
+      sinon.stub(repo.git, 'reset').throws();
+
+      assert.deepEqual(reporterProxy.addEvent.callCount, 0);
+      try {
+        await repo.undoLastCommit();
```

Comment:
If you want to be _sure_ this is a failure, you can replace this try-catch with:

```js
await assert.isRejected(repo.undoLastCommit());
```

Reply:
I _do_ want to be sure! :) 

Thanks for all the great suggestions for cleaning up test assertions @smashwilson. I like the improved cleanliness and readability ✨